pkg/controller/logcollector/logcollector_controller.go

```diff
@@ -405,36 +409,49 @@ func (r *ReconcileLogCollector) Reconcile(ctx context.Context, request reconcile
 		return reconcile.Result{RequeueAfter: 5 * time.Second}, nil
 	}
 
-	esgwCertificate, err := certificateManager.GetCertificate(r.client, relasticsearch.PublicCertSecret, common.OperatorNamespace())
-	if err != nil {
-		r.status.SetDegraded(operatorv1.ResourceValidationError, fmt.Sprintf("Failed to retrieve / validate  %s", relasticsearch.PublicCertSecret), err, reqLogger)
-		return reconcile.Result{}, err
-	} else if esgwCertificate == nil {
-		log.Info("Elasticsearch gateway certificate is not available yet, waiting until they become available")
-		r.status.SetDegraded(operatorv1.ResourceNotReady, "Elasticsearch gateway certificate are not available yet, waiting until they become available", nil, reqLogger)
+	// Determine whether or not this is a multi-tenant management cluster.
+	multiTenantManagement := r.multiTenant && managementCluster != nil
+	if instance.Spec.MultiTenantManagementClusterNamespace != "" && !multiTenantManagement {
+		r.status.SetDegraded(operatorv1.ResourceValidationError, "multiTenantManagementClusterNamespace can only be set on multi-tenant management clusters", nil, reqLogger)
 		return reconcile.Result{}, nil
 	}
 
-	// The location of the Linseed certificate varies based on if this is a managed cluster or not.
+	// The name of the Linseed certificate varies based on if this is a managed cluster or not.
 	// For standalone and management clusters, we just use Linseed's actual certificate.
-	linseedCertLocation := render.TigeraLinseedSecret
+	linseedCertName := render.TigeraLinseedSecret
+	linseedCertNamespace := common.OperatorNamespace()
+	var tenant *operatorv1.Tenant
 	if managedCluster {
 		// For managed clusters, we need to add the certificate of the Voltron endpoint. This certificate is copied from the
 		// management cluster into the managed cluster by kube-controllers.
-		linseedCertLocation = render.VoltronLinseedPublicCert
+		linseedCertName = render.VoltronLinseedPublicCert
+	} else if multiTenantManagement {
+		// For multi-tenant management clusters, the linseed certificate isn't in the tigera-operator namespace.
+		// Instead, look for a Tenant instance that represent's the management cluster's own tenant.
+		if instance.Spec.MultiTenantManagementClusterNamespace == "" {
+			r.status.SetDegraded(operatorv1.ResourceValidationError, "multiTenantManagementClusterNamespace is not set", nil, reqLogger)
+			return reconcile.Result{}, nil
+		}
+		tenant, _, err = utils.GetTenant(ctx, r.multiTenant, r.client, instance.Spec.MultiTenantManagementClusterNamespace)
+		if err != nil {
+			r.status.SetDegraded(operatorv1.ResourceNotReady, fmt.Sprintf("Failed to retrieve tenant in ns %s", instance.Spec.MultiTenantManagementClusterNamespace), err, reqLogger)
+			return reconcile.Result{}, err
+		}
```

Comment:
Isn't tenant.Namespace alsways equal to instance.Spec.MultiTenantManagementClusterNamespace?

Reply:
yeah, this check is to confirm that a real Tenant CR actually exists - it doesn't expect the value to be any different.

Reply:
It took me  a little while to understand this block of code. If we would move the code to get the tenant above this if-else block, I think  it could improve the readability.

Reply:
I'm not sure I understand the suggestion. I don't think we can move the tenant querying code because we need to query for the tenant referenced in the `instance.Spec.MultiTenantManagementClusterNamespace`. To do that, we must first check that field is actually set.

We only want to query the tenant and check that field if this is indeed a multi-tenant management cluster. Which sort of places some limitations on where we can actually perform this query.

Maybe a better comment explaining all of this would help?

Reply:
I moved it around a little bit and I think improved the comment. LMK what you think